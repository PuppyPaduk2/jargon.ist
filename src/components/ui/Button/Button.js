// @flow

import React, { type Node } from 'react';
import cx from 'classnames';

import './Button.scss';

type Props = {
  children: Node,
  primary?: boolean,
  className?: string,
};

const Button = ({ primary, className, children, ...otherProps }: Props) => (
  <button className={cx('c-button', { 'c-button--primary': primary }, className)} {...otherProps}>
    {children}
  </button>
);

Button.defaultProps = {
  primary: false,
  className: '',
};

export default Button;
