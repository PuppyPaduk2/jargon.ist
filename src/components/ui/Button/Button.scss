@import '../../../../node_modules/buttono/buttono';
@import '../../../styles/abstracts/index';

$buttono-bg-color: $primary-color;
$buttono-bg-hover: scale-color($buttono-bg-color, $lightness: $buttono-function-factor);
$buttono-radius: px-to-rem(4px);
$buttono-line-height: (20 / 14);
$buttono-padding-ends: px-to-rem(15px);
$buttono-padding-sides: px-to-rem(48px);
$buttono-font-family: $g-font-secondary;

.c-button {
  @include buttono-block;

  &--primary {
    @include buttono-style-modifier($border-color: null, $border-hover-color: null);
  }
}
