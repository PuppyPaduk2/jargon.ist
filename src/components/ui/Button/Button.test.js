// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

it('renders correctly default button', () => {
  const wrapper = shallow(<Button>Example button</Button>);

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly primary button', () => {
  const wrapper = shallow(<Button primary>Example button</Button>);

  expect(wrapper).toMatchSnapshot();
});
