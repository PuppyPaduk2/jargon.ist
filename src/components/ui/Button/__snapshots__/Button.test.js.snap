// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly default button 1`] = `
<button
  className="c-button"
>
  Example button
</button>
`;

exports[`renders correctly primary button 1`] = `
<button
  className="c-button c-button--primary"
>
  Example button
</button>
`;
